src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -476,10 +476,9 @@ internal void CleanupSubscriptions()
 
                 var recentConnection = state.Value.MostRecentEndedConnection();
 
-                while (recentConnection != null && recentConnection.Stats.LastMessageSentAt < oldestPossibleIdleSubscription)
+                if (recentConnection != null && recentConnection.Stats.LastMessageSentAt < oldestPossibleIdleSubscription)
```

Comment:
this will now only kill a single one? Why would this code cause a hang?

Reply:
@ayende we are trying to kill useless SubscriptionConnection objects, not "recent connections".  and recent connections list is limited to 10 anyway. In case of real low memory, a more through cleanup will be performed.
This code _caused_ a hang because recentConnection was never replaced and it didn't try to  remove items from the recent connections list.
The point in this cleanup is to make sure there are no unused subscription objects in the SubscriptionStorage.